webui/src/lib/onboarding/repoOnboardingService.ts

```diff
@@ -0,0 +1,119 @@
+import { OnBoardingSteps } from "./types";
+import { NavigateFunction } from "react-router-dom";
+import { branches, retention, branchProtectionRules, objects } from "../api";
+
+const IMPORT_BRANCH_REGEX = /^_(.*)_imported$/;
+const HOOKS_PATH = "_lakefs_actions/";
+const SAMPLE_REPO_FILE_NAME = "lakes.parquet";
+
+// generate a random file name to avoid collisions for the upload preflight check
+const generateRandomFileName = (): string => {
+  const randomString = Math.random().toString(36).substring(2, 15);
```

Comment:
Why 15?  AFAICT this only gives 10 or 11 characters

Reply:
This includes `0.`. The length of the string is somewhere between 12 and 14, depending on how close the number is to 1.

Reply:
I feel like we're saying the same thing but reaching two different conclusions, and I'm a bit confused

I haven't done log base conversions in a long time, and the last time I did, it was more like Matlab did it.  
Let's say `Math.random` returned 0.006619485802314751 (that's something that happens, apparently). 
If you convert that to a base36 string, you'll get `0.08ku6zrn6a46`, which is 14 characters.

`substring` includes the start index and excludes the end index, so that's from position 2 to 14 of the original string.
No? Is my math off? Am I missing the obvious? 😊